lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -270,11 +271,21 @@ def _close_indices_if_needed(self, backup_id: int) -> Set[str]:
         return indices_to_close
 
     def _restore(self, backup_id: int) -> Dict[str, Any]:
-        """Runs the restore and processes the response."""
+        """Runs the restore and processes the response.
+
+        Backups come in the user-formatted string, YYYY-MM-DDTHH:MM:SSZ.
+        """
         backup_indices = self._list_backups()[backup_id]["indices"]
+        # now, fix the id format for the actual request
+        id = self._format_backup_id(
+            backup_id,
+            format_from=BACKUP_ID_FORMAT,
+            format_to=OPENSEARCH_BACKUP_ID_FORMAT,
```

Comment:
Why do we do this?

Reply:
I have converged both formats, the only remark is that opensearch only accepts minor chars as part of its snapshot ID. Therefore, we have to `lower()` and `upper()` in certain points of the code (as `time.time` uses capital letters and they are more visually easier).